base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -174,20 +191,37 @@ pub fn create_ordered_transaction_hashes_from_block(block: &monero::Block) -> Ve
         .collect()
 }
 
-/// Appends merge mining hash to a Monero block
-pub fn append_merge_mining_tag<T: AsRef<[u8]>>(block: &mut monero::Block, hash: T) -> Result<(), MergeMineError> {
+/// Inserts merge mining hash into a Monero block
+pub fn proxy_insert_merge_mining_tag_into_block<T: AsRef<[u8]>>(
```

Comment:
```suggestion
pub fn insert_merge_mining_tag_into_block<T: AsRef<[u8]>>(
```
Same here, remove the proxy name, or something that tells the users what makes it special